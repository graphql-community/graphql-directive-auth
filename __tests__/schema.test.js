const { graphql } = require('graphql');
const directive = require('../src/index');
const MockExpressRequest = require('mock-express-request');
const schema = require('../example/schema');

test('getDirectiveDeclaration should be defined', () => {
  expect(directive('123').getDirectiveDeclaration()).toMatchSnapshot();
});

test('if call resolver if Authorization header is set to correct value', () =>
  graphql(
    schema,
    `
      query {
        me
      }
    `,
    {},
    {
      req: new MockExpressRequest({
        headers: {
          Authorization:
            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbG8iOjEyMywiaWF0IjoxNTMyNzI3MDMzfQ.nSGh5q9YXZd-pWYagOxD2f-9hdNPw08e4MnKASgn4wY',
        },
      }),
    }
  ).then(response => {
    expect(response.data.me).toBeTruthy();
  }));

test('if throw an Error if Authorization header is not set to correct value', () =>
  graphql(
    schema,
    `
      query {
        me
      }
    `,
    {},
    {
      req: new MockExpressRequest({ headers: {} }),
    }
  ).then(response => {
    expect(response.errors[0].message).toEqual('Not authorized!');
    expect(response.data.me).toBeNull();
  }));
