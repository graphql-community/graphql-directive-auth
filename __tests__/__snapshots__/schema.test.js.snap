// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDirectiveDeclaration should be defined 1`] = `
GraphQLDirective {
  "args": Array [],
  "astNode": undefined,
  "description": undefined,
  "locations": Array [
    "FIELD_DEFINITION",
  ],
  "name": "isAuthenticated",
}
`;
